sae_lens/training/activations_store.py

```diff
@@ -451,27 +455,42 @@ def get_activations(self, batch_tokens: torch.Tensor):
 
         n_batches, n_context = batch_tokens.shape
 
-        stacked_activations = torch.zeros((n_batches, n_context, 1, self.d_in))
+        # For some models, we might want to exclude some positions from the sequence to train on
+        training_context_slice = list(
```

Comment:
This is inefficient to create a list of indices like this. It's better to create an actual `slice` object, like `training_context_slice = slice(self.start_pos_offset, n_context - self.end_pos_offset)`. Alternative, you can probably just do something like the following directly when selecting the specific indices.

```python
stacked_activations[:, :, 0] = layerwise_activations[self.hook_name][
      :, start_pos:end_pos, self.hook_head_index
]
```

Reply:
Hey David, thanks for your thorough review of our PR! As Joseph mentioned in his [comment](https://github.com/jbloomAus/SAELens/pull/317#issuecomment-2391999612) above, we've rebased on Callum's PR and added a few tests of our own. These problems are not present in the rebased code, but they are really helpful for @zhenningdavidliu and me! 🙏